src/NJsonSchema.Tests/Generation/FlattenInheritanceHierarchyTests.cs

```diff
@@ -155,5 +164,25 @@ public async Task When_JsonSchemaFlattenAttribute_is_used_on_class_then_inherite
             Assert.True(schema.Definitions["B"].Properties.ContainsKey("Bbb"));
             Assert.True(schema.Definitions["B"].Properties.ContainsKey("Ccc"));
         }
+
+
+        [Fact]
+        public async Task When_class_inherited_and_json_flattened_then_ignore_base_property_with_same_name()
+        {
+            //// Arrange
+            var settings = new JsonSchemaGeneratorSettings
+            {
+                FlattenInheritanceHierarchy = true,
+            };
+
+            //// Act & Assertion
+            var schema = JsonSchema.FromType(typeof(Teacher), settings);
+            var data = schema.ToJson();
+
+            // if there's a definition it correctly parsed the data
+            Assert.True(schema.Definitions.ContainsKey("Schedule"));
+            Assert.True(schema.Properties["Schedule"].Type.HasFlag(JsonObjectType.Null));
+            Assert.True(schema.Properties["Schedule"].Type.HasFlag(JsonObjectType.Array));

```

Comment:
Maybe you should also check whether the Type.Items is of type String or HasReference (class) - just express/test what you expect in the case of Teacher (class) and Person (string), no?

Reply:
Sorry, but I don't think I follow, when you're referring to `Type.Items` there's no access for `Items` if I try to access it from the statement `schema.Properties["Schedule"].Type`.

The only thing that makes sense to me, is to use `schema.Properties["Schedule"].Item.HasReference` as it will mean this is a class object vs string. I don't know the API well enough to see what the underline schema type is class/string

Reply:
Ah yes it is `schema.Properties["Schedule"].Item.HasReference` vs `Item.Type == string` we should check

Reply:
Ah I understand now, Would you like me to add 1 more test to specifically test that it's "string" when I do `FromType(typeof(Person))` to verify it works as intended? Or can I extend the test *When_FlattenInheritanceHierarchy_is_enabled_then_all_properties_are_in_one_schema* to include an assertion for `schema.Properties["Schedule"].Item.Type == JsonObjectType.String `

Reply:
For now extended the current test case to include person as well to test the types. I believe this will give a meaningful test as you described earlier. Let me know if there's any issues with this! I really appreciate it.